_posts/criteo/2020-03-10-DRAFT-contextualbandits.md

```diff
@@ -73,11 +77,15 @@ We thus note, for a policy $\pi$,  $\pi(a,x)$ the probability of choosing action
 
 When training models on a contextual bandit problem, the goal is to find the policy which maximizes the average reward:
 
-$$ Argmax_{ \pi } \mathbb{E}_X ( \mathbb{E}_{ A \sim \pi } ( \mathbb{E}( R | A = a , X = x ))) $$
+$$\hat{\pi} = Argmax_{ \pi } \mathbb{E}_X ( \mathbb{E}_{ A \sim \pi } ( \mathbb{E}( R | A = a , X = x ))) $$
+
+Note that the optimal policy $\hat{ \pi}$ is usually deterministic. ( In each context, just choose the action which maximize the expected reward $\mathbb{E}( R \| X=x,  A=a) $ <!-- please check my change -->
 
-Note that the optimal policy $\hat{ \pi}$ is usually deterministic. ( In each context, just choose the action which maximize the expected reward $\mathbb{E}( X \| X=x,  A=a) $
+However, it is usually a good idea to avoid fully deterministic policies. One of the main reasons is that a randomized policy allows us to keep some exploration on the different actions, and this is useful to learn how to improve the policy. It is also useful to evaluate a new policy, as we will describe in the next sections.
 
-However, it is usually a good idea to avoid fully deterministic policies. One of the main reasons is that a randomized policy allows to keep some exploration on the different actions, and this is useful to learn how to improve the policy. It is also useful to evaluate a new policy, as we will describe in the next sections.
+<!--
+^ this is a fairly subtle point..  I think we use policies because they are differentiable and to satisfy the IPS formula - but deterministic makes most sense to me.  issues around uncertainty either the notion of some kind of portfolio or explore exploit are separate..   I dont know how to do it better than you have.. but maybe we can discuss...
```

Comment:
well, in practice the first reason why we randomized is because it brings some diversity o the user